lib/ansible/modules/monitoring/pagerduty_event.py

```diff
@@ -0,0 +1,370 @@
+        required: false
+    client_url:
+        description:
+        -  The URL of the monitoring client that is triggering this event.
+        required: false
+    custom_details:
+        description:
+            - Additional details about the event and affected system.
+            - key, value pairs of data to load into the incident.
+        required: false
+        aliases: [ details ]
+'''
+
+EXAMPLES = '''
+# Trigger an incident with just the basic options
+- pagerduty_incident:
+    routing_key: xxx
+    state: triggered
+    severity: warning
+    desc: problem that led to this trigger
+    source: host.example.com
+  register: new_incident
+
+# Print incident key generated by PagerDuty API v2
+- debug:
+    # For Events API v1 use: "{{ new_incident['response']['incident_key'] }}"
+    msg: "{{ new_incident['response']['dedup_key'] }}"
+
+# Trigger an incident with more options
+- pagerduty_incident:
+    routing_key: xxx
+    state: triggered
+    severity: critical
+    desc: problem that led to this trigger
+    dedup_key: "{{ 'zzz' | hash('md5') }}"
+    source: "{{ ansible_hostname }}"
+    client: Sample Monitoring Service
+    client_url: http://service.example.com
+    custom_details:
+      ping_time: "1500ms"
+      load_avg: "0.75"
+
+# Acknowledge an incident based on incident_key
+- pagerduty_incident:
+    routing_key: xxx
+    state: acknowledged
+    dedup_key: somekey
+
+# Resolve an incident based on incident_key
+- pagerduty_incident:
+    routing_key: xxx
+    state: resolved
+    dedup_key: somekey
+'''
+
+RETURN = '''
+api_version:
+    description: PagerDuty Event API version.
+    returned: changed
+    type: string
+payload:
+    description: Output JSON payload provided to module.
+    returned: changed
+    type: dict
+status:
+    description: Output HTTP status code.
+    returned: success
+    type: string
+response:
+    description: Output HTTP response body.
+    returned: success
+    type: complex
+    contains:
+        message:
+            description: Response message.
+            type: string
+            returned: success
+            sample:
+                - "Event processed"
+        status:
+            description: Response status code.
+            type: string
+            returned: success
+            sample:
+                - "202"
+        dedup_key:
+            description: Response incident key - Events API v1 returns C(incident_key) and v2 returns C(dedup_key).
+            type: string
+            returned: success
+            sample:
+                - "somekey"
+'''
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        routing_key=dict(type='str', required=True, aliases=['service_key']),
+        state=dict(type='str', required=True,
```

Comment:
these states are a little confusing...  granted I am not using pagerduty day to day anymore.

Is there a way to use our commonly used states (present, absent, etc)?

Reply:
Thanks for the comments @thaumos! The PagerDuty Events API (v1 and v2) accepts the actions: `trigger`, `acknowledge` and `resolve` for event state. Hence, they do not correlate to the common Ansible module states.

I can change the current mapping of `triggered`, `acknowledged` and `resolved` to the slightly more shorthand version if required.

Reply:
Updated to use the shorthand states of `trigger`, `acknowledge` and `resolve` @thaumos